src/algorithm/radix3.rs

```diff
@@ -153,6 +153,15 @@ impl<T: FftNum> Radix3<T> {
             layer_twiddles = &layer_twiddles[twiddle_offset..];
         }
     }
+
+    fn perform_fft_out_of_place(
```

Comment:
IMO radix3, radix4, and radixn should all be consistent here: Either all of their OOP should just forward to the immutable version, or all of their OOP should have a dedicated implementation. Since radix4 and radixn have dedicated implementations, I vote for giving this its dedicated implementation back.